contracts/AaveYield.sol

```diff
@@ -8,32 +8,65 @@ import './libraries/Errors.sol';
 import './interactors/AaveInteractor.sol';
 import './YieldStorage.sol';
 import './interfaces/IAaveYield.sol';
+import './WardenHandler.sol';
 
-contract AaveYield is UUPSUpgradeable, Ownable2StepUpgradeable, AaveInteractor, YieldStorage, IAaveYield {
-  /// @notice initialize function used during contract deployment
-  /// @param aavePool address of a Aave pool
-  /// @param tokens array with addresses of tokens which will be used in the Aave pool
-  function initialize(address aavePool, address[] calldata tokens) external initializer {
-    __Ownable_init(msg.sender);
-    __UUPSUpgradeable_init();
-    __AaveInteractor_init(aavePool, tokens);
+contract AaveYield is
+  UUPSUpgradeable,
+  Ownable2StepUpgradeable,
+  AaveInteractor,
+  YieldStorage,
+  IAaveYield,
+  WardenHandler
+{
+  // /// @notice initialize function used during contract deployment
+  // /// @param aavePool address of a Aave pool
+  // /// @param tokens array with addresses of tokens which will be used in the Aave pool
+  // function initialize(address aavePool, address[] calldata tokens) external initializer {
+  //   __Ownable_init(msg.sender);
+  //   __UUPSUpgradeable_init();
+  //   __AaveInteractor_init(aavePool, tokens);
+  // }
+
+  function initializeV2(
+    address underlyingToken,
+    address axelarGateway,
+    address axelarGasService,
+    string calldata wardenChain,
+    string calldata wardenContractAddress
+  ) external reinitializer(2) {
+    __AaveInteractor_initV2(underlyingToken);
+    __WardenHandler_init(axelarGateway, axelarGasService, wardenChain, wardenContractAddress);
+
+    // TODO: add lpAmount totalSupply initial value
   }
 
   /// @dev method called during the contract upgrade
   function _authorizeUpgrade(address newImplementation) internal override onlyOwner {}
 
   /// @inheritdoc IAaveYield
-  function stake(address token, uint256 amount, string calldata userWardenAddress) external returns (uint256 shares) {
+  function stake(uint64 stakeId, uint256 amount) external returns (uint256 shares) {
+    require(msg.sender == address(this));
```

Comment:
If only the contract itself can call the stake and unstake methods, why are they external?